rpcs3/Emu/Cell/PPUTranslator.cpp

```diff
@@ -1787,6 +1787,12 @@ void PPUTranslator::ADDIS(ppu_opcode_t op)
 {
 	Value* imm = m_ir->getInt64(op.simm16 << 16);
 
+	if (m_rel && m_rel->type == 5)
+	{
+		imm = m_ir->CreateShl(ZExt(ReadMemory(GetAddr(+2), GetType<u16>()), GetType<u64>()), 16);
+		m_rel = nullptr;
+	}
+
 	if (m_rel && m_rel->type == 6)
```

Comment:
else if

Reply:
Solved with a better solution.

Reply:
type 5 on ADDIS didn't require ZExt instead of SExt?

Reply:
was a copypasta typo.